packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -91,6 +92,9 @@ Future<XcodeBuildResult> buildXcodeProject({
     return XcodeBuildResult(success: false);
   }
 
+  final IOSProjectMigration migration = IOSProjectMigration(app.project, globals.logger);
+  migration.migrate();
```

Comment:
Couldn't we pretty easily use the `.metadata` file to keep track of migrations that we've done (say, with a UUID for each completed migration)?

In addition to efficiency over time, it also means that we won't accidentally fight users. E.g., if someone had their own unrelated reasons that they needed `disable_input_output_paths` in their podspec, they'd have a really bad time if we re-run on every build.

Reply:
I really wish we could wait until https://docs.google.com/document/d/1fPnpUI6gXE9PkNWuFoAW20m_sQ87ZSC1rrn_D7fxBvs/edit?ts=5e58625e#heading=h.pub7jnop54q0 was in place and use that instead of having a bunch of one-off migrators fighting each other... but we can't wait in this case.

Reply:
Since I removed the `disable_input_output_paths` change, I didn't implement migration versioning, but it really is a good idea.  I'd like to split it out into it's own change as part of the wider `flutter migrate` effort.  For now it's running on every run like https://github.com/flutter/flutter/pull/13011.